src/openforms/submissions/signals.py

```diff
@@ -57,3 +59,22 @@ def delete_submission_report_files(
     logger.debug("Deleting file %r", instance.content.name)
 
     instance.content.delete(save=False)
+
+
+@receiver(submission_complete, dispatch_uid="submission.increment_form_counter")
+def increment_form_counter(sender, instance: Submission, **kwargs):
+    submitted_form = instance.form
+
+    try:
+        existing_form_statistics = FormStatistics.objects.get(form=submitted_form)
+    except FormStatistics.DoesNotExist:
+        FormStatistics.objects.create(
+            form=submitted_form,
+            form_name=submitted_form.name,
+            submission_count=1,
+            last_submission=timezone.now(),
+        )
+    else:
+        existing_form_statistics.submission_count += 1
+        existing_form_statistics.last_submission = timezone.now()
+        existing_form_statistics.save()
```

Comment:
Maybe using [`get_or_create`](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#get-or-create)?

Reply:
Nice!This is part of Django 5.0, so in 4.2 `F` expressions are not supported directly in both `get_or_create` and `create_or_update` methods (I want to increment the counter), so I will then have to do an extra call to update separately the counter.